PySCF.py

```diff
@@ -80,6 +80,33 @@ def is_consistent(self,other):
             .format(key,self.__dict__[key],other.__dict__[key]))
         return False
     return True
+
+  #---------------------------------------------------
+  def update_input(self, fname):
+    # check if input file already updated
+    lines = open(fname, 'r').read().split()
```

Comment:
The PySCF writer doesn't need to know about which file to write to. The file management is not done at that level. It only needs to know that it is reading from a given check file.

This capability was already there in the pyscf writer. You need only to change the recipe to enable restarts, not the pyscf writer.